source/server/lds_api.cc

```diff
@@ -75,13 +76,16 @@ void LdsApiImpl::onConfigUpdate(const ResourceVector& resources, const std::stri
         ENVOY_LOG(debug, "lds: add/update listener '{}' skipped", listener_name);
       }
     } catch (const EnvoyException& e) {
-      throw EnvoyException(
-          fmt::format("Error adding/updating listener {}: {}", listener_name, e.what()));
+      exception_msgs.push_back(fmt::format("{}: {}", listener_name, e.what()));
     }
   }
 
   version_info_ = version_info;
   runInitializeCallbackIfAny();
+  if (!exception_msgs.empty()) {
+    throw EnvoyException(fmt::format("Error adding/updating listener(s) {}",
+                                     StringUtil::join(exception_msgs, ", ")));
```

Comment:
Why do we join with `, ` here and https://github.com/envoyproxy/envoy/pull/5806/files does `\n` for CDS? Ideally they look and behave as close to eah other as possible.

FWIW, this also relates to https://github.com/envoyproxy/envoy/issues/6026.

Reply:
I thought comma separated is more concise rather than a `\n`. I was actually thinking of changing CDS to to be like this once this is merged. But if you think I can change this to `\n` - not a strong preference. Also CDS is not adding the line `Error adding/updating cluster(s)` which I think we should add.

Reply:
@htuch found another problem with CDS implementation - it does not include the cluster name, it is difficult to find which cluster is the problematic one. Fixed and changed it also to use the same `,` separated join in https://github.com/envoyproxy/envoy/pull/6039

Reply:
I left a comment on #6039, I will continue reviewing this PR to move forward, but I feel we're introducing some technical debt.